src/lv_widgets/lv_msgbox.c

```diff
@@ -198,6 +198,26 @@ void lv_msgbox_set_text(lv_obj_t * mbox, const char * txt)
     mbox_realign(mbox);
 }
 
+/**
+ * Set a formatted text for the message box
+ * @param mbox pointer to a message box
+ * @param fmt `printf`-like format
+ */
+void lv_msgbox_set_text_fmt(lv_obj_t * mbox, const char * fmt, ...)
+{
+    LV_ASSERT_OBJ(mbox, LV_OBJX_NAME);
+    LV_ASSERT_STR(fmt);
+
+    lv_msgbox_ext_t * ext = lv_obj_get_ext_attr(mbox);
+
+    va_list args;
+    va_start (args, fmt);
+    lv_label_set_text_fmt(ext->text, fmt, args);
```

Comment:
Your problem is that `va_list` doesn't work this way. `...` and a `va_list` can't be used interchangeably.

There are two solutions here:

* Copy all the logic for handling the label string from `lv_label_set_text_fmt`. I wouldn't recommend this; we'd end up with two almost identical implementations and it will make the function twice as long.
* Split out [this chunk](https://github.com/lvgl/lvgl/blob/79433deeb8680fe9e72c9992e81d25c58a11a91d/src/lv_widgets/lv_label.c#L277-L320) of `lv_label_set_text_fmt` into a new function (`lv_label_set_text_vfmt`) which basically takes a `va_list` instead of the ellipsis (`...`). Then you could use that function from here. This is the more sustainable solution in my opinion.